### ISSUE TITLE
Linux-Installer adb is outdated

### ISSUE THREAD

**Issue by opensource21** [koush/support-wiki#93]
I tried Helium at Wileyfox Spark+. The desktop installer doesn't find the device. With adb 1.0.32 I can connect to the device. Unfortunately I was unable to adjust the `run.sh`.
It set `BASE=/usr/local/bin`, but the script end as the shell and doesn't exit and Helium isn't enabled.
If I do it manually it's work, but it's difficult to exit.

Any ideas how to make further investigations to fix this issue?

**Comment by curtisy1** [comment]
Depending on what Distro you are using try installing ncurses5, I don't know if it works on Debian based systems but for Arch I know that this can't be fixed probably until he recompiles that adb for ncurses6, as the problem is in there. If you have Ubuntu or something, tell me how it went please so I can maybe do it myself.

**Comment by opensource21** [comment]
To make it clear, I use the command line script. I can connect with the newer adb version to the shell of the wileyfox. If I do every step of the shell-script manually it works, only if I run run.sh it fails. I can't see how ncursed could fix the issue.

**Comment by curtisy1** [comment]
Yeah I know you are talking about the command line script, there would be no use in using run.sh otherwise. I also think we mean the same thing in general. What I was trying to say is that the script is depending on an old and outdated version of adb (which requires ncurses5 to work), so it will error out when running it without modifications. Now there are three solutions to this problem:
1. The developer updates this thing and uses the newest adb version (as you confirmed it works), which I doubt he will as this project seems pretty dead to me.
2. Your method of basically adjusting the script and make it work with the new adb (this is difficult, you said yourself).
or 3. My way of installing ncurses5 which the old adb depends on (probably just as difficult because I don't know if it's possible to use both 6 and 5 at the same time on debian distributions. On arch it's possible using ncurses5-compat-libs and adjusting names likely; debian should be backwardscompatible nowadays so this shouldn't be an issue as well.) which should in theory let the script run without error using the old adb.
When I find the time I might mess with this a bit on a VM and see how far I can get. I'd be glad if you could maybe help on that as well.

**Comment by opensource21** [comment]
Ok now I understand your point. With Ubuntu 16.04 I can reproduce your problem. I would recommend to use `BASE=$(dirname $(which adb))` or remove `$BASE/` at `$BASE/adb shell << EOF`. This ensures that you use always the newest possible version on your computer.
Painful that the script isn't supported anymore. 
The problem I have seems to be wileyfox specific.

**Comment by bhavin192** [comment]
On Fedora 29 the script gives 
```console
$ ./run.sh 
./run.sh: line 2: ./adb: No such file or directory
```
Even though, the adb executable is there, trying to run `./adb` gives same error, not sure why.

As @opensource21 said solution for me was to install `android-tools` using `dnf` and then changing the `BASE` variable.

```console
$ sudo dnf install android-tools
```
